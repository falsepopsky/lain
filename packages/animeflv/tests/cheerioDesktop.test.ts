import { browse, browseQuery, browseQueryPageTen, fake } from '../fixtures/index.js';
import { getPages, getTitles } from '../src/index.js';

describe('getTitles()', () => {
  describe('Fixture browse', () => {
    const titles = getTitles(browse);

    it('Should return titles', () => {
      expect(Array.isArray(titles)).toBe(true);
      expect(titles).toHaveLength(24);
      expect(titles).toMatchSnapshot();
    });

    it('Should return a string type for every value of the keys from titles', () => {
      for (const title of titles) {
        expect(typeof title.title).toBe('string');
        expect(typeof title.description).toBe('string');
        expect(typeof title.slug).toBe('string');
        expect(typeof title.type).toBe('string');
        expect(typeof title.cover).toBe('string');
      }
    });

    it('Should have the expected keys in the data object', () => {
      for (const title of titles) {
        expect(title).toHaveProperty('title');
        expect(title).toHaveProperty('description');
        expect(title).toHaveProperty('slug');
        expect(title).toHaveProperty('type');
        expect(title).toHaveProperty('cover');
      }
    });
  });

  describe('Fixture browse-query', () => {
    const titles = getTitles(browseQuery);

    it('Should return titles', () => {
      expect(Array.isArray(titles)).toBe(true);
      expect(titles).toHaveLength(7);
      expect(titles).toMatchSnapshot();
    });

    it('Should return a string type for every value of the keys from titles', () => {
      for (const title of titles) {
        expect(typeof title.title).toBe('string');
        expect(typeof title.description).toBe('string');
        expect(typeof title.slug).toBe('string');
        expect(typeof title.type).toBe('string');
        expect(typeof title.cover).toBe('string');
      }
    });

    it('Should have the expected keys in the data object', () => {
      for (const title of titles) {
        expect(title).toHaveProperty('title');
        expect(title).toHaveProperty('description');
        expect(title).toHaveProperty('slug');
        expect(title).toHaveProperty('type');
        expect(title).toHaveProperty('cover');
      }
    });
  });

  describe('Fixture browse-query-page-10', () => {
    const titles = getTitles(browseQueryPageTen);

    it('Should return titles', () => {
      expect(Array.isArray(titles)).toBe(true);
      expect(titles).toHaveLength(24);
      expect(titles).toMatchSnapshot();
    });

    it('Should return a string type for every value of the keys from titles', () => {
      for (const title of titles) {
        expect(typeof title.title).toBe('string');
        expect(typeof title.description).toBe('string');
        expect(typeof title.slug).toBe('string');
        expect(typeof title.type).toBe('string');
        expect(typeof title.cover).toBe('string');
      }
    });

    it('Should have the expected keys in the data object', () => {
      for (const title of titles) {
        expect(title).toHaveProperty('title');
        expect(title).toHaveProperty('description');
        expect(title).toHaveProperty('slug');
        expect(title).toHaveProperty('type');
        expect(title).toHaveProperty('cover');
      }
    });
  });

  describe('Fixture fake', () => {
    const titles = getTitles(fake);

    it('Should return empty strings for slug & cover if not found', () => {
      // @ts-expect-error: Object is possibly 'undefined'.
      expect(titles[0].slug).toBe('');
      // @ts-expect-error: Object is possibly 'undefined'.
      expect(titles[0].cover).toBe('');
    });
  });
});

describe('getPages()', () => {
  describe('Fixture browse', () => {
    const pages = getPages(browse);

    it('Returns an array containing the pages', () => {
      expect(Array.isArray(pages)).toBe(true);
      expect(pages).toHaveLength(11);
    });

    it('should return an array of unique pages', () => {
      const uniquePages = Array.from(new Set(pages));

      expect(uniquePages).toEqual(pages);
    });
  });

  describe('Fixture browse-query', () => {
    const pages = getPages(browseQuery);

    it('Returns an array containing the pages', () => {
      expect(Array.isArray(pages)).toBe(true);
      expect(pages).toHaveLength(1);
    });

    it('should return an array of unique pages', () => {
      const uniquePages = Array.from(new Set(pages));

      expect(uniquePages).toEqual(pages);
    });
  });

  describe('Fixture browse-query-page-10', () => {
    const pages = getPages(browseQueryPageTen);

    it('Returns an array containing the pages', () => {
      expect(Array.isArray(pages)).toBe(true);
      expect(pages).toHaveLength(11);
    });

    it('should return an array of unique pages', () => {
      const uniquePages = Array.from(new Set(pages));

      expect(uniquePages).toEqual(pages);
    });
  });
});
