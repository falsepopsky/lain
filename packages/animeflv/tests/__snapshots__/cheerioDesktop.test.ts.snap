// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTitles() Fixture browse Should return titles 1`] = `
[
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3816.jpg",
    "description": "",
    "slug": "/anime/black-clover-mahou-tei-no-ken",
    "title": "Black Clover: Mahou Tei no Ken",
    "type": "Película",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3815.jpg",
    "description": "",
    "slug": "/anime/5toubun-no-hanayome-movie",
    "title": "5-toubun no Hanayome Movie",
    "type": "Película",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3813.jpg",
    "description": "",
    "slug": "/anime/kimi-wo-aishita-hitori-no-boku-e",
    "title": "Kimi wo Aishita Hitori no Boku e",
    "type": "Película",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3814.jpg",
    "description": "",
    "slug": "/anime/boku-ga-aishita-subete-no-kimi-e",
    "title": "Boku ga Aishita Subete no Kimi e",
    "type": "Película",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3812.jpg",
    "description": "Una conspiración de larga data se arremolina sobre un poder misterioso conocido como la "Reina" de Raja, un pequeño país al oeste de Tempest. Cuando un slime que se convirtió en un Demon Lord llamado Rimuru Tempest se cruza con Hiiro, un sobreviviente de la raza de los Ogros, comienza una increíble aventura repleta de nuevos personajes. ¡El p...",
    "slug": "/anime/tensei-shitara-slime-datta-ken-movie-guren-no-kizunahen",
    "title": "Tensei shitara Slime Datta Ken Movie: Guren no Kizuna-hen",
    "type": "Película",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3811.jpg",
    "description": "En un reino en el que los humanos han de servir al rey de los demonios, Sariphi es ofrecida como sacrificio. Sin embargo, con su carisma no tardará en conquistar al rey de los demonios.",
    "slug": "/anime/niehime-to-kemono-no-ou",
    "title": "Niehime to Kemono no Ou",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3810.jpg",
    "description": "",
    "slug": "/anime/watashi-ni-tenshi-ga-maiorita-precious-friends",
    "title": "Watashi ni Tenshi ga Maiorita! Precious Friends",
    "type": "Película",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3809.jpg",
    "description": "",
    "slug": "/anime/ousama-ranking-yuuki-no-takarabako",
    "title": "Ousama Ranking: Yuuki no Takarabako",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3806.jpg",
    "description": "Cuarta temporada de Majutsushi Orphen Hagure Tabi",
    "slug": "/anime/majutsushi-orphen-hagure-tabi-seiikihen",
    "title": "Majutsushi Orphen Hagure Tabi: Seiiki-hen",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3808.jpg",
    "description": "La chica de 16 años, Ai Hoshino, es una talentosa y hermosa idol que es adorada por sus fanáticos. Ella es la personificación de la pureza, pero no todo lo que brilla es oro… Gorou Honda es un ginecólogo de una zona rural y gran fanático de Ai. Entonces, cuando la idol embarazada se presenta en su hospital, está más que desconcertado. Goro...",
    "slug": "/anime/oshi-no-ko",
    "title": "[Oshi No Ko]",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3807.jpg",
    "description": "Uzuki Shimamura es una aspirante a estudiante que sueña con ser una idol. Tras fracasar en su primera audición, un productor de 346 Productions le comunica a Uzuki que ha sido elegida para el “Proyecto Cenicienta”, un nuevo grupo de ídolos. Sin embargo, al proyecto le faltan otros dos miembros, por lo que el productor necesita buscar candida...",
    "slug": "/anime/the-idolmster-cinderella-girls-u149",
    "title": "The IDOLM@STER Cinderella Girls: U149",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3805.jpg",
    "description": "",
    "slug": "/anime/the-marginal-service",
    "title": "The Marginal Service",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3803.jpg",
    "description": "El momento de un duelo de proporciones épicas: el héroe legendario contra el poderoso nigromante conocido como Dios de los Cadáveres. Cuando el combate llega a su fin, algo falla. Mientras, en otro mundo se despierta un niño llamado Polka Shinoyama. ¿Qué es ese mundo? ¿Qué ese cuerpo en el que está? Nadie podría haber esperado que tan ép...",
    "slug": "/anime/dead-mount-death-play",
    "title": "Dead Mount Death Play",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3802.jpg",
    "description": "Eunha Park pensó que tenía otra oportunidad en la vida cuando despertó en el mundo de una novela. ¿El único problema? ¡Raeliana McMillan, el personaje en el que renació, está destinada a ser asesinada por su prometido! Por el bien de su supervivencia, se le ocurre una idea: contraer un matrimonio falso con un hombre más poderoso que su fut...",
    "slug": "/anime/kanojo-ga-koushakutei-ni-itta-riyuu",
    "title": "Kanojo ga Koushaku-tei ni Itta Riyuu",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3804.jpg",
    "description": "Todo comenzó con la muerte de uno de los miembros del Club de Astronomía. A partir de esto, el rumoreado fantasma del club cobró vida. Ganta Nakami, insomne y gruñón, se encuentra con Isaki Magari, moderadamente rebelde pero fácil de llevar, en el observatorio abandonado del Club de Astronomía. Magari se despierta de golpe y trata de escapar...",
    "slug": "/anime/kimi-wa-houkago-insomnia",
    "title": "Kimi wa Houkago Insomnia",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3801.jpg",
    "description": "Tercera temporada de Kimetsu no Yaiba",
    "slug": "/anime/kimetsu-no-yaiba-katanakaji-no-satohen",
    "title": "Kimetsu no Yaiba: Katanakaji no Sato-hen",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3800.jpg",
    "description": "Kokona es una joven chica que suena con convertirse en actriz de la compañía SIRIUS, una de las más importantes que hay.",
    "slug": "/anime/world-dai-star",
    "title": "World Dai Star",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3799.jpg",
    "description": "Había una vez un héroe que fue convocado a otro mundo, y salvó ese mundo. Sin embargo, el hombre cayó en una "trampa" y fue devuelto a la fuerza a su mundo original. Para colmo, tuvo que empezar de nuevo como un bebé... ¡Esta es la historia de un alocado viaje por otro mundo en el que un antiguo héroe que se reencarnó en un estudiante un po...",
    "slug": "/anime/isekai-shoukan-wa-nidome-desu",
    "title": "Isekai Shoukan wa Nidome desu",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3798.jpg",
    "description": "Desde que su padre murió, Ichirou Kuga ha luchado para mantener a sus dos hermanos menores con nada más que una pequeña herencia y su pasión por dibujar manga. Pero cada vez es más difícil mantenerse al día con sus crecientes responsabilidades y plazos, especialmente después de que sus dos últimos asistentes renunciaron para seguir sus sue...",
    "slug": "/anime/otonari-ni-ginga",
    "title": "Otonari ni Ginga",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3789.jpg",
    "description": "Asahi Ikusaba es un chico cualquiera de preparatoria que acaba por accidente en un mundo mágico de fantasía. Esto no deprime a Asahi, quien se dispone a usar sus conocimientos de los videojuegos y de los isekai para disfrutar de ese nuevo mundo, pero, para su desgracia, las habilidades que tiene son de bajísimo nivel. Si eso no fuera suficiente,...",
    "slug": "/anime/isekai-one-turn-kill-neesan-ane-douhan-no-isekai-seikatsu-hajimemashita",
    "title": "Isekai One Turn Kill Neesan: Ane Douhan no Isekai Seikatsu Hajimemashita",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3792.jpg",
    "description": "Segunda temporada de Tonikaku Kawaii",
    "slug": "/anime/tonikaku-kawaii-2nd-season",
    "title": "Tonikaku Kawaii 2nd Season",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3793.jpg",
    "description": "Tousuke Rokudou es un estudiante de primer año que asiste al Instituto Aomori, una escuela llena de delincuentes. A menudo es acosado por sus compañeros de clase debido a su incompatibilidad con su comportamiento, siendo las únicas excepciones sus dos buenos amigos que también son marginados sociales. Frustrados por la forma en que son tratados...",
    "slug": "/anime/rokudou-no-onnatachi",
    "title": "Rokudou no Onna-tachi",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3794.jpg",
    "description": "Hayato Kasukabe ha sido aceptado en la Universidad de Tokio en su primer intento. Al recibir la noticia de la muerte de su abuela, regresa a la casa de su infancia, el Café Terrace Familia, por primera vez en tres años para encontrar allí a cinco extrañas chicas que dicen ser “la familia de la abuela”. ¡La inesperada vida de Hayato en un p...",
    "slug": "/anime/megami-no-cafe-terrace",
    "title": "Megami no Café Terrace",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3791.jpg",
    "description": "Liza Luna es una extraterrestre que quiere destruir la Tierra nada más verla. Pero el planeta tiene un montón de cosas buenas, y sería una pena que ella no lo comprobara. Así, la Tierra se salva de la destrucción, y la alienígena empieza a vagar por los lugares. Pronto, Liza acaba en un café para gatos, donde se encontrará con criaturas que...",
    "slug": "/anime/kawaisugi-crisis",
    "title": "Kawaisugi Crisis",
    "type": "Anime",
  },
]
`;

exports[`getTitles() Fixture browse-query Should return titles 1`] = `
[
  {
    "cover": "https://animeflv.net/uploads/animes/covers/1488.jpg",
    "description": "Bajo el constante ataque de los Angeles, NERV introduce dos nuevos pilotos: la misteriosa Makinami Mari Illustrious, y a la intensa Asuka Langley Shikinami. Paralelamente, Gendo Ikari y SEELE proceden con el proyecto secreto que involucra tanto a Rei como a Shinji. You Can (Not) Advance comienza con el ataque del Tercer Ángel y la aparici&oa...",
    "slug": "/anime/evangelion-20-you-can-not-advance",
    "title": "Evangelion: 2.22 You Can (Not) Advance",
    "type": "Película",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/1494.jpg",
    "description": "El final original de Neon Genesis Evangelion mostrado en los dos últimos capítulos de la serie de TV fue criticado unánimemente debido a la enorme cantidad de preguntas que dejó sin resolver. Sin más remedio, GAINAX decidió rehacer el final lanzando la película The End of Evangelion (El Final de Evan...",
    "slug": "/anime/neon-genesis-evangelion-the-end-of-evangelion",
    "title": "Neon Genesis Evangelion: The End of Evangelion",
    "type": "Película",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3509.jpg",
    "description": "En el año 2015, el mundo está al borde de su destrucción. La última esperanza de la humanidad reside en las manos de NERV, una agencia especial bajo el mando de las Naciones Unidas, y sus Evangelions, unos robots gigantes capaces de derrotar a los Ángeles que anuncian la ruina de la Tierra. Gendou Ikari, jefe de la organización, busca pilotos...",
    "slug": "/anime/evangelion-30-10-thrice-upon-a-time",
    "title": "Evangelion 3.0+1.0 Thrice Upon a Time",
    "type": "Película",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/1487.jpg",
    "description": "El argumento está basado en los episodios del 1 al 6 de la serie original. Aunque algunas situaciones se muestran de diferente forma, tales como ángulos de cámara, historias alternativas, etc. Shinji Ikari llega a Tokio-3, buscado por su padre Gendo. A su llegada, se encuentra con ejercito de la ONU tratando de detener al cuart...",
    "slug": "/anime/evangelion-10-you-are-not-alone",
    "title": "Evangelion: 1.11 You Are (Not) Alone",
    "type": "Película",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/460.jpg",
    "description": "Según cuentan los libros de historia, el 13 de Septiembre del 2000, un enorme meteorito chocó contra la Antártida, causando el derretimiento del Polo Sur y la consecuente inundación y destrucción de todas las ciudades costeras. A este evento crucial se lo denomino Segundo Impacto -El primero fue el que destruy&oac...",
    "slug": "/anime/neon-genesis-evangelion",
    "title": "Neon Genesis Evangelion",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/1493.jpg",
    "description": "Es la primera película de la serie de televisión Neon Genesis Evangelion. Consiste en dos partes: Death y Rebirth, Muerte y Renacimiento respectivamente. La primera parte, Death, consiste de una edición de 60 minutos sobre los primeros 24 episodios del programa de televisión, que incluye varias escenas inéditas. L...",
    "slug": "/anime/neon-genesis-evangelion-death-and-rebirth",
    "title": "Neon Genesis Evangelion: Death & Rebirth",
    "type": "Película",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/1019.jpg",
    "description": "La historia empieza 14 años después del Casi Tercer Impacto, el gobierno japonés decide desmantelar NERV y arrestar a su personal, durante los 14 años varios miembros de NERV obtienen su libertad y deciden formar una organización llamada WILL-E cuyo supuesto propósito es limpiar los mares, en realidad trata...",
    "slug": "/anime/evangelion-30-you-can-not-redo",
    "title": "Evangelion: 3.0 You Can (Not) Redo",
    "type": "Película",
  },
]
`;

exports[`getTitles() Fixture browse-query-page-10 Should return titles 1`] = `
[
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3729.jpg",
    "description": "Con la intención de proteger su pueblo natal del Imperio de Erebonia y para limpiar su nombre de la mancha que supuso la deslealtad de su abuelo, Lavie se alista en los Jaegers del Norte, donde hará todo lo que esté en su mano para cumplir su misión. Ahora esta joven que no se amolda a las normas deberá formar un escuadrón para embarcarse en ...",
    "slug": "/anime/the-legend-of-heroes-sen-no-kiseki-northern-war",
    "title": "The Legend of Heroes: Sen no Kiseki - Northern War",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3361.jpg",
    "description": "El Imperio, dotado de un nivel avanzado de conocimiento, y la Agencia de la Casa Imperial de Nebulis temida como el «País de las Brujas», mantienen una lucha que dura en el tiempo. En el campo de batalla un chico y una chica se conocen. El espadachín que se convirtió en la fuerza definitiva en el campo de batalla a la edad más joven a lo larg...",
    "slug": "/anime/kimi-to-boku-no-saigo-no-senjou-aruiwa-sekai-ga-hajimaru-seisen",
    "title": "Kimi to Boku no Saigo no Senjou, Aruiwa Sekai ga Hajimaru Seisen",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/681.jpg",
    "description": "Megaman no es un anime en sí, las ovas existentes están basadas en un videojuego llamado Rockman (en japón} Megamán (aqui en occidente). Es la historia de un robot llamado X el cual pelea contra los Maveriks que son robots que se han opuesto a los humanos y se dedican a destruir, X junto con Zero, otro riploid están dedicados a proteger la paz...",
    "slug": "/anime/megaman-maverick-hunter-x-the-day-of-sigma",
    "title": "MegaMan Maverick Hunter X The Day of Sigma",
    "type": "OVA",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/384.jpg",
    "description": "Continuan las aventuras de Keima Katsuragi, el rey de las conquistas (apodo dado por ser como un dios en juegos Galge de citas 2D), seguirá "capturando" el corazón de chicas con espiritus prófugos dentro de ellas para que puedan salir y Elsee, su compañera demoniaca, les de captura.",
    "slug": "/anime/kami-nomi-zo-shiru-sekai-2",
    "title": "Kami Nomi zo Shiru Sekai 2",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3469.jpg",
    "description": "Mi amiga de la infancia, Shida Kuroha, parece tener sentimientos románticos hacia mí. Ella es también mi vecina, y es pequeña y linda. Con una hermosa personalidad, ella es como mi hermana mayor, y esa es una de sus principales fortalezas… Pero yo ya tengo a alguien que me gusta, la hermosa idol de la escuela, Kachi Shirokusa. Aunque pensánd...",
    "slug": "/anime/osananajimi-ga-zettai-ni-makenai-love-comedy",
    "title": "Osananajimi ga Zettai ni Makenai Love Comedy",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3672.jpg",
    "description": "Hace mucho tiempo, había un Dios de la Luz y un Dios de las Tinieblas. Mientras que los devotos del Dios de la Luz experimentaban felicidad y buena fortuna, el Dios de las Tinieblas engañaba a la gente y les robaba la alegría. Y así, como dice la leyenda, el Dios de la Luz decidió castigarlo convirtiéndolo en un monstruo. Enfurecido por esto,...",
    "slug": "/anime/totsukuni-no-shoujo-2022",
    "title": "Totsukuni no Shoujo (2022)",
    "type": "OVA",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3401.jpg",
    "description": "Noir, el tercer hijo de una familia de nobles en decadencia, ha perdido su empleo y no sabe qué hacer, pero la fortuna le sonríe justo cuando decide convertirse en un aventurero. Calabozos a los que nadie sabe cómo entrar. Lugares legendarios llenos de extremadamente raras criaturas, herramientas mágicas con poderes inmensurables y objetos q...",
    "slug": "/anime/ore-dake-haireru-kakushi-dungeon",
    "title": "Ore dake Haireru Kakushi Dungeon",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3675.jpg",
    "description": "¡Las villanas tienen su propia ruta! Cuando su compromiso con el príncipe finalizó de forma poco cortés, la joven noble Aileen comenzó a recordar poco a poco detalles de su vida anterior y se percató de algo: está viviendo en el mundo de uno de sus juegos otome favoritos, ¡y como la villana principal! Pero su memoria es un desastre y no rec...",
    "slug": "/anime/akuyaku-reijou-nanode-last-boss-wo-kattemimashita",
    "title": "Akuyaku Reijou nanode Last Boss wo Kattemimashita",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/1393.jpg",
    "description": "Kosuke Ueki, un chico normal que vive en Japón, vive una vida normal, amigos normales... pero él se siente diferente. Un día, un nuevo profesor, el profesor Kobayashi, se interesa en Ueki tras saber que él lo que más desea es justicia en el mundo. Tras esto, le revela a Ueki su verdadera identidad; Kobayashi es...",
    "slug": "/anime/the-law-of-ueki",
    "title": "The Law of Ueki",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/2117.jpg",
    "description": "Segunda temporada de The IDOLM@STER: Cinderella Girls",
    "slug": "/anime/the-idolmaster-cinderella-girls-2",
    "title": "The IDOLM@STER: Cinderella Girls 2nd Season",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/2331.jpg",
    "description": "",
    "slug": "/anime/strike-the-blood-valkyria-no-oukoku-hen",
    "title": "Strike the Blood: Valkyria no Oukoku-hen",
    "type": "OVA",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/2428.jpg",
    "description": "Segunda temporada de Concrete Revolutio.",
    "slug": "/anime/concrete-revolutio-choujin-gensou-the-last-song",
    "title": "Concrete Revolutio: Choujin Gensou - The Last Song",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3084.jpg",
    "description": "Por mucho que los niños la quieran, ella no es su madre. Viven juntos, pero no son familiares de sangre. Grace Field es un hogar al que son enviados los niños que no tienen padres. Un hogar irreemplazable para los 38 niños que allí viven felices su día a día, como si fueran hermanos, aunque al inicio fueran completos desconocidos. Al menos as...",
    "slug": "/anime/yakusoku-no-neverland",
    "title": "Yakusoku no Neverland",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/67.jpg",
    "description": "Amores lejanos, mensajes de aliento, pequeñas aventuras, secretos inconfesables... Pocos objetos pueden reunir tantos sentimientos y revelaciones en tan poco espacio como una carta. Desde su velada creación hasta la ansiosa espera de su llegada, es cierto que somos personas distintas antes y después de leer una carta: tal ve...",
    "slug": "/anime/shigofumi",
    "title": "Shigofumi",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/292.jpg",
    "description": "La serie tiene lugar en el Japón del año 2029, tras el alza de este país como imperio económico debido al "milagro japonés" (reciclaje de residuos nucleares). La historia gira en torno a la organización de fuerzas especiales Sección 9, y en particular en torno a su segunda de abordo tras Da...",
    "slug": "/anime/ghost-in-the-shell-stand-alone-complex",
    "title": "Ghost in the Shell: Stand Alone Complex",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/853.jpg",
    "description": "La historia se sitúa en un Japón de dentro de mil años. Cinco niños (Saki la protagonista, Satoru, Maria, Mamoru y Shun) han nacido y se han criado en un tranquilo pueblo del cual se podría decir que es una utopía, con agua abundante y verdes zonas. El mundo de esa época está gobernado por aquellos con el ?poder maldito?, también conocido ...",
    "slug": "/anime/shinsekai-yori",
    "title": "Shinsekai yori",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/1732.jpg",
    "description": "",
    "slug": "/anime/detective-conan-movie-18",
    "title": "Detective Conan Movie 18: The Sniper from Another Dimension",
    "type": "Película",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/2546.jpg",
    "description": "",
    "slug": "/anime/danganronpa-3-the-end-of-kibougamine-gakuen-kibou-hen",
    "title": "Danganronpa 3: The End of Kibougamine Gakuen - Kibou-hen",
    "type": "OVA",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/2771.jpg",
    "description": "La historia de Ballroom e Youkoso se centra en el estudiante Tatara Fujita, que quiere ser bueno en algo, en cualquier cosa. Por desgracia, es tan mediocre como cualquier otro adolescente. Los matones del barrio lo saben y por eso aprovechan cualquier ocasión para sacarle dinero, pero esto cambia cuando el caballeroso Kaname Sengoku se encuentra c...",
    "slug": "/anime/ballroom-e-youkoso",
    "title": "Ballroom e Youkoso",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/2829.jpg",
    "description": "Chise Hatori, 15 años. Perdida, sin esperanza y sin familia, la compran por dinero. Pero quien la compra no es otra persona, sino un hechicero no humano llamado Elias. Aunque al inicio duda, la joven comenzará una nueva vida como su aprendiz y futura esposa. Se muda y comienza a vivir una nueva vida pacífica, tranquila pero firme, hasta que un d...",
    "slug": "/anime/mahoutsukai-no-yome",
    "title": "Mahoutsukai no Yome",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/392.jpg",
    "description": "La historia toma lugar en Tokyo Butei High School, una escuela especial en donde detectives armados llamados "Butei" son entrenados en el uso de sus armas y habilidades. Kinji Tooyama es un estudiante de segundo año que tiene una habilidad especial, pero la mantiene en secreto y se muestra como alguien normal ya que quiere una vida...",
    "slug": "/anime/hidan-no-aria",
    "title": "Hidan no Aria",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/3407.jpg",
    "description": "El anime se centra en Reki, una estudiante de segundo año de secundaria a quien le encanta andar en patineta. Un día, el se ve atrapado en “S”, una peligrosa carrera de patinetas sin reglas que se realiza en mina abandonada. Langa, quien ha regresado a Japón tras vivir algún tiempo en Canadá y que nunca antes había patinado en patinet...",
    "slug": "/anime/sk",
    "title": "SK∞",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/1484.jpg",
    "description": "La historia transcurre en la Era Bakumatsu, cuando los cimientos del gobierno de Tokugawa empezaron a tambalearse. En la edad donde la "Oscuridad" se respiraba y la presencia de monstruos era muy temida, un escritor soñador, Yamaoka Hyakusuke, viaja por todos los países con la esperanza de publicar en todo el mundo su ob...",
    "slug": "/anime/kousetsu-hyaku-monogatari",
    "title": "Kousetsu Hyaku Monogatari",
    "type": "Anime",
  },
  {
    "cover": "https://animeflv.net/uploads/animes/covers/1494.jpg",
    "description": "El final original de Neon Genesis Evangelion mostrado en los dos últimos capítulos de la serie de TV fue criticado unánimemente debido a la enorme cantidad de preguntas que dejó sin resolver. Sin más remedio, GAINAX decidió rehacer el final lanzando la película The End of Evangelion (El Final de Evan...",
    "slug": "/anime/neon-genesis-evangelion-the-end-of-evangelion",
    "title": "Neon Genesis Evangelion: The End of Evangelion",
    "type": "Película",
  },
]
`;
